DEV: Add to-markdown plugin api

CvX opened the pull request:
To be used in https://github.com/discourse/discourse-spoiler-alert/pull/85

jjaffeux commented on app/assets/javascripts/discourse/app/lib/plugin-api.js:
````diff
@@ -1680,6 +1684,38 @@ class PluginApi {
+   * });
+   * ```
+   */
+  beforeToMarkdownTagDecorate(callback) {
````
What you call tag is what we call inline at other places right?

CvX replied:
In most cases - I think so? But since `Tag` is the base class for all tags, its decorate method is called for e.g. `blockquote`, `td`, `th` - so also elements that are not strictly inline.

tgxworld commented on app/assets/javascripts/discourse/app/lib/plugin-api.js:
````diff
@@ -1680,6 +1684,38 @@ class PluginApi {
     addSectionLink(arg);
   }
+
+  /**
+   * Decorate inline bbcode in to-markdown function.
+   *
+   * ```
+   * api.beforeToMarkdownTagDecorate(function (text) {
+   *   if (this.element.attributes.class === "spoiled") {
+   *     this.prefix = "[spoiler]";
+   *     this.suffix = "[/spoiler]";
+   *   }
+   * });
+   * ```
+   */
````
Just some general sentiments but after reading the description here, I can't really tell what purpose the API serves and what I can do with this API as an end user.

tgxworld commented on app/assets/javascripts/discourse/app/lib/plugin-api.js:
```diff
@@ -1680,6 +1684,38 @@ class PluginApi {
   }
+
+  /**
+   * Decorate inline bbcode in to-markdown function.
```
I think we're referring to the `toMarkdown` function in the `to-markdown` file? Also, I'm not sure if this is tied to `bbcode` because I can technically insert an `prefix` or `suffix` which are not bbcode?

tgxworld commented:
I think it'll be helpful if we have tests for the APIs added.

CvX commented:
@tgxworld I added tests, and changed the description/examples. The api here isn't too user friendly, since it's pretty much just a way to inject code into these two specific functions in to-markdown. The only place that needs this is discourse-spoiler-alert, so this API is effectively just for that.

tgxworld commented:
> The api here isn't too user friendly, since it's pretty much just a way to inject code into these two specific functions in to-markdown. The only place that needs this is discourse-spoiler-alert, so this API is effectively just for that.

I wonder if we should just do away with adding it as an plugin api in this case since only one plugin requires it and we happen to be maintainers of that plugin and that plugin's tests are run as part of our CI. The reason I say so because I see `plugin-api` as our client side public facing API and if the API isn't user friendly, I see very little value it adding the API as part of `plugin-api`.
